src/source.rs

```diff
@@ -332,6 +340,7 @@ mod tests {
                             partition_id: 0,
                         },
                         keys: vec![],
+                        headers:Arc::clone(&shared_headers), // Cloning the Arc, not the HashMap,
```

Comment:
```suggestion
                        headers:Arc::clone(&shared_headers),
```

no need of that comment since we can clearly see it is `Arc::clone`